json_yaml_files/prepare_files.py

```diff
@@ -0,0 +1,178 @@
+from bids import BIDSLayout
+from collections import defaultdict
+import re
+import json
+import sys
+import yaml
+from pathlib import Path
+
+
+class BIDStoNDAConfigGenerator:
+    def __init__(self):
+        self.path = sys.argv[1]
+        self.layout = BIDSLayout(self.path, derivatives=True)
+        self.A = 'image03'
+        self.pattern = r'^(?P<A>[^_]+)_(?P<X>[^.]+)\.(?P<Y>[^.]+)\.(?P<Z>[^.]+)\..+$'
+        # self.top_level_files = ['CHANGES',
+        #                         'README', 'dataset_description.json']
+        self.scan_type = {
+            'anat': {
+                'mprage': 'MR structural (MPRAGE)',
+                't1w': 'MR structural (T1)',
+                'pd': 'MR structural (PD)',
+                'fspgr': 'MR structural (FSPGR)',
+                'fsip': 'MR structural (FISP)',
+                't2w': 'MR structural (T2)',
+                'pd_t2': 'MR structural (PD, T2)',
+                'b0_map': 'MR structural (B0 map)',
+                'b1_map': 'MR structural (B1 map)',
+                'flash': 'MR structural (FLASH)',
+                'mp2rage': 'MR structural (MP2RAGE)',
+                'tse': 'MR structural (TSE)',
+                't1w_t2w': 'MR structural (T1, T2)',
+                'mpnrage': 'MR structural (MPnRAGE)'
+            },
+            'pet': {
+                'pet': 'PET'
+            }
+        }
+        self.image_modality = {
+            'pet': 'PET',
+            'anat': 'MRI',
+            'func': 'MRI',
+            'dwi': 'MRI',
+            'fmap': 'MRI'
+        }
+
+    def extract_file_name_components(self, file_name):
+        match = re.match(self.pattern, file_name)
+        if match:
+            A = match.group('A')
+            X = match.group('X')
+            Y = match.group('Y')
+            Z = match.group('Z')
+            return A, X, Y, Z
+        else:
+            return None
+
+    def prepare_json_contents(self, file_name):
+        # json_contents = {file: file for file in self.top_level_files}
+        json_contents = {}
+        _, X, Y, _ = self.extract_file_name_components(file_name)
+        files = self.layout.get(
+            scope=X if X != 'inputs' else 'raw', datatype=Y, return_type='file')
+        for file_path in files:
+            sub_index = file_path.find('derivatives')
+            if sub_index == -1:
+                sub_index = file_path.find('sub-')
+            if sub_index != -1:
```

Comment:
Should this just be an `else` since it is the complement?

```suggestion
            else:
```